### PR TITLE
Update documentation on ``pyqt4v2``

### PR DESCRIPTION
**Pull request by montefra**
I have a tentative update of the documentation according to what we have discussed in pytest-dev/pytest-qt#148.

I'm not very satisfied by it. I might change some of it in the next days. Comments are welcome

### CODE REVIEW

File: docs/intro.rst [hunk 1]
```diff
@@ -40,19 +40,39 @@ this order:
...
     [pytest]
     qt_api=pyqt5
 
-
 Alternatively, you can set the ``PYTEST_QT_API`` environment
 variable to the same values described above (the environment variable wins over the configuration
 if both are set).
 
-.. _version2: http://pyqt.sourceforge.net/Docs/PyQt4/incompatible_apis.html
+Starting with ``pytest-qt`` version 2.0, ``PyQt`` or ``PySide`` are loaded after
```
Comment (nicoddemus) [inline docs/intro.rst]: Probably this is too much information to be into the "introduction" part of the docs. Perhaps we can add an entry "A note about pyqt4v2", similar to [A note about QApplication.exit()](http://pytest-qt.readthedocs.io/en/latest/app_exit.html)?

File: docs/intro.rst [hunk 2]
```diff
@@ -40,19 +40,39 @@ this order:
...
+the ``conftest.py`` files but before all the test modules. This can lead to some
+unexpected behaviour if ``pyqt4v2``.
```
Comment (nicoddemus) [inline docs/intro.rst]: Probably missing a "if pyqt4v2 is set" or something like this.

**Comment by nicoddemus** [comment]
Thanks for the follow up @montefra!

File: docs/intro.rst (hunk 2 above)
Comment (montefra) [inline docs/intro.rst]: right. Fixed

File: docs/intro.rst [hunk 3]
```diff
@@ -40,19 +40,39 @@ this order:
...
     qt_api=pyqt5
 
-
 Alternatively, you can set the ``PYTEST_QT_API`` environment
 variable to the same values described above (the environment variable wins over the configuration
 if both are set).
 
-.. _version2: http://pyqt.sourceforge.net/Docs/PyQt4/incompatible_apis.html
+Starting with ``pytest-qt`` version 2.0, ``PyQt`` or ``PySide`` are loaded after
+the ``conftest.py`` files but before all the test modules. This can lead to some
```
Comment (montefra) [inline docs/intro.rst]: Much better than mine version. A new commit is arriving

**Comment by montefra** [comment]
@nicoddemus I've updated the info according to your suggestions.

----

on a slightly related note: would you mind if I add ``extras_require={'doc': ['sphinx', 'sphinx_rtd_theme']},`` in the ``setup`` call and update the Development information to use ``pip install -e .`` or ``pip install -e .[doc]`` , instead of ``python setup.py develop``?
I found myself using a couple of different venvs and having to run ``make html``,  ``pip install sphinx``, ``make html`` and then ``pip install sphinx_rtd_theme`` to be able to build the documentation locally.

**Comment by nicoddemus** [comment]
> on a slightly related note: would you mind if I add extras_require={'doc': ['sphinx', 'sphinx_rtd_theme']}, in the setup call and update the Development information to use pip install -e . or pip install -e .[doc] , instead of python setup.py develop?

Not at all, seems like a much better approach than the current one. :grin:

**Comment by montefra** [comment]
can I do it here or would you prefer a new branch?

**Comment by nicoddemus** [comment]
Feel free to do it here, just make sure to use a separate commit. :+1:

**Comment by nicoddemus** [comment]
Excellent, many thanks @montefra!

**Comment by montefra** [comment]
Glad to contribute back to a tool I use regularly.
Thank you for accepting the PR.

**Pull request merged**